### DISCUSSION TITLE
Does klog automatically create new files?

### DISCUSSION THREAD

**Discussion by MuellerKlaus** [jotaen/klog#304] [Help / Question]
*Background:*
I cloned klog source with Windows, compiled with go (go version go1.20.1 windows/amd64) and started to test tracking.

*Problem:*
All tests showed an error about missing file. I expected that 'klog track testtask test.klg' creates a new file if not exists. I see this as default behavior of such tools that a file is used if exists or created if missing

Is this expected that a user need to create new files in advance ('touch test.klog') of use it or is this a Windows issue? I do not find any specific explanation about manual file creation in any documentation.

**Comment by jotaen** [comment]
This isn’t a Windows-specific issue, klog generally doesn’t create any new files. So it’s expected that you’d have to create them manually before being able to use them.

It would theoretically be possible to introduce some sort of “create on the fly” mechanism, although this might be unexpected in other ways, because if you’d accidentally misspell a file name, then klog wouldn’t fail anymore.

**Reply by MuellerKlaus** [reply]
Would it make sense to add this info to one of the documentations? It was hard to understand this behavior as first time user.

**Reply by jotaen** [reply]
Sure, good idea! I’ve adjusted the initial description of the [“Manipulate files” section](https://klog.jotaen.net/#manipulate).

**Reply by QuAzI** [reply]
If the file name already used in config (like default file) it should be automatically created. Why not?
For example, the file name is `time_tracking.klg` and each month moved to `time_tracking-YYYY-MM.klg` (archived) after reports generation.

**Reply by QuAzI** [reply]
Or please add config option to autocreate file

**Comment by jotaen** [comment]
I’m open to reconsider this behaviour, though I’m still a bit reluctant to make it the default. Maybe one approach could be to provide a config option in `config.ini`? E.g.:

```
auto_create_files = true
```

**Reply by QuAzI** [reply]
`auto_create_files = true` as option would be nice
